### DISCUSSION TITLE
Module to manage ad banners?

### DISCUSSION THREAD

**Discussion by LITUATUI** [hbstack/theme-cards#256] [Q&A] [answered]
When I was using Wordpress, I used a plug-in called [Ad Inserter Pro](https://adinserter.pro/) which was very helpful to manage ad banners (rotate, slide and show alternative content when ad blocker detected).

Is there a module I can use to rotate/slide banners? Even better if it can also show alternative banner to Ad Sense code if ad blocker is detected?

Thanks!

**Comment by razonyang** [comment]
I do have a _private_ anti adblock module, but it's super simple, which just pop up an dialog if ad blocker detected.

![image](https://github.com/hbstack/theme-cards/assets/17720932/3b5a57be-9df2-4d3f-b8c2-aea777b585b2)

The alternative content is a good idea, will see how to implement it, you can describe those features in detail if you're insterested in it. However I'm busy on other stuff (no much time on programing) in next one or two weeks, this feature won't happen too fast.

Btw, that module won't be public, I'll give private repos access to sponsors, but GitHub Sponsors allows only one private repos permission for sponsors, however I wish give sponsors multiple repos permission, that's why there is no docs for the anti adblock module.

**Reply by LITUATUI** [reply]
Ignoring the ad blocking feature, is there a module that rotates/slide images with links to external websites?

I would like to rotate/slide ads of my sponsors on some hooks:

- header
- after post
- footer
- sidebar

Rotating banners helps to reduce ad blindness/fatigue.

**Reply by razonyang** [reply, marked as answer]
> is there a module that rotates/slide images with links to external websites?

No, there isn't yet.

But you can easilly create your own with Bootstrap [Carousel](https://getbootstrap.com/docs/5.3/components/carousel/) and the theme's hooks.

**Reply by razonyang** [reply]
Btw, you can get inspired by the `hb-custom` module located in `params.yaml` to customize hooks.

The hooks definitions can be found on the docs site.

1. Header: https://hbstack.dev/modules/header/overview/#hooks
2. Blog: https://hbstack.dev/modules/blog/hooks/
3. Footer: https://hbstack.dev/modules/footer/hooks/

**Reply by LITUATUI** [reply]
This worked great, thanks!
